storage/src/nodestore.rs

```diff
@@ -1601,13 +1616,95 @@ impl<S: WritableStorage> NodeStore<Committed, S> {
     }
 }
 
+pub struct FreeListIterator<'a, S: ReadableStorage> {
```

Comment:
pub(crate)? Otherwise, this is exposed to the outside interface. This might be what we want, but let's be clearer about what is exposed and what isn't.